server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -883,7 +883,7 @@ public int getNumberOfReplicas() {
     }
 
     public boolean isSegrepEnabled() {
-        return isSegrepEnabled;
+        return Boolean.TRUE.equals(isSegrepEnabled);
```

Comment:
nit - this isn't required bc we always initialize this variable in the constructor.
`isSegrepEnabled = settings.getAsBoolean(IndexMetadata.SETTING_SEGMENT_REPLICATION, false);`

Reply:
True but leaving it in to convert to boolean which is faster since we don't need a thread safe implementation.